### PR TITLE
[IMPORTANT] update pyjwt (might cause vulnerabilities!)

### PR DESCRIPTION
**Pull request by RikoSmith**
(I purposely made it a PR instead of a push to get these changes reviewed)

This PR updates the` pyjwt` (`1.7.1` -> `2.3.0`) package but with some nuances:

- `pyjwt` had changes in `encode` and `decode` and I made changes to make tests pass, without the full context of the code written. It's possible we might do the same changes in our backend code too.
- there are many controversies around `pyjwt` as many authors face issues with the new version and the majority resort to downgrading to previous versions (https://github.com/flavors/django-graphql-jwt/issues/242, https://github.com/jazzband/djangorestframework-simplejwt/issues/346, https://github.com/jazzband/djangorestframework-simplejwt/pull/349).

### CODE REVIEW

File: libdrf/login/settings.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
     'JWT_SECRET_KEY': settings.SECRET_KEY,
     'JWT_ALGORITHM': 'HS256',
-    'JWT_VERIFY': True,
+    'JWT_VERIFY': False,
```
Comment (adsva) [inline libdrf/login/settings.py]: Is this change really needed? I think this just sets the default value for verifying, and I think it's good if that is True?

File: setup.py [hunk 2]
```diff
@@ -17,7 +17,7 @@
     install_requires=[
         'djangorestframework>=3.8',
         'requests>=2',
-        'PyJWT==1.7.1',
+        'PyJWT>=1.7.1',
```
Comment (adsva) [inline setup.py]: I think perhaps it's better to pin this to ~=2 So it will always upgrade to a version above 2?

**Review by adsva — CHANGES_REQUESTED** [review]

**Comment by RikoSmith** [comment]
@adsva turns out `verify=True` is not an argument anymore and it was causing the error (https://pyjwt.readthedocs.io/en/stable/changelog.html#dropped-deprecated-verify-param-in-jwt-decode). Instead it should be passed into `options` dict argument. Adjusted the code:)
